packages/m2m-gateway/src/services/tenantService.ts

```diff
@@ -1,8 +1,46 @@
+import { WithLogger } from "pagopa-interop-commons";
+import { m2mGatewayApi } from "pagopa-interop-api-clients";
 import { PagoPAInteropBeClients } from "../clients/clientsProvider.js";
+import { M2MGatewayAppContext } from "../utils/context.js";
+import { toM2MTenant } from "../api/tenantApiConverter.js";
 
 export type TenantService = ReturnType<typeof tenantServiceBuilder>;
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function tenantServiceBuilder(_clients: PagoPAInteropBeClients) {
-  return {};
+export function tenantServiceBuilder({
+  tenantProcessClient,
+}: PagoPAInteropBeClients) {
+  return {
+    getTenants: async (
+      { logger, headers }: WithLogger<M2MGatewayAppContext>,
+      queryParams: m2mGatewayApi.GetTenantsQueryParams
```

Comment:
Minor, for consistency, context is usually the last param we pass

```suggestion
      queryParams: m2mGatewayApi.GetTenantsQueryParams,
      { logger, headers }: WithLogger<M2MGatewayAppContext>
```